crates/nu-command/src/conversions/into/filesize.rs

```diff
@@ -110,28 +108,43 @@ fn into_filesize(
     )
 }
 
-pub fn action(input: &Value, span: Span) -> Value {
-    match input {
-        Value::Filesize { .. } => input.clone(),
-        Value::Int { val, .. } => Value::Filesize { val: *val, span },
-        Value::Float { val, .. } => Value::Filesize {
-            val: *val as i64,
-            span,
-        },
-        Value::String { val, .. } => match int_from_string(val, span) {
-            Ok(val) => Value::Filesize { val, span },
-            Err(error) => Value::Error { error },
-        },
-        _ => Value::Error {
+pub fn action(input: &Value) -> Value {
+    if let Ok(value_span) = input.span() {
+        match input {
+            Value::Filesize { .. } => input.clone(),
+            Value::Int { val, .. } => Value::Filesize {
+                val: *val,
+                span: value_span,
+            },
+            Value::Float { val, .. } => Value::Filesize {
+                val: *val as i64,
+                span: value_span,
+            },
+            Value::String { val, .. } => match int_from_string(val, value_span) {
+                Ok(val) => Value::Filesize {
+                    val,
+                    span: value_span,
+                },
+                Err(error) => Value::Error { error },
+            },
+            _ => Value::Error {
+                error: ShellError::UnsupportedInput(
+                    "'into filesize' for unsupported type".into(),
+                    value_span,
+                ),
+            },
+        }
+    } else {
+        Value::Error {
             error: ShellError::UnsupportedInput(
-                "'into filesize' for unsupported type".into(),
-                span,
+                "unable to find span on 'into filesize' value".into(),
+                Span::test_data(),
```

Comment:
I'd use the span you're getting into the function up above, `span`. The `Span::test_data()` command should only be used when writing test data, but here we're trying to show a real error to the user.

Reply:
I should not you removed that bit, but it'd be good to bring it back and use it here

Reply:
ok, i re-undid-removed-put-back span again ;)